commonlib/src/main/java/com/streamsets/pipeline/lib/parser/excel/Cells.java

```
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 StreamSets Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file
+ */
+
+package com.streamsets.pipeline.lib.parser.excel;
+
+import com.streamsets.pipeline.api.Field;
+import com.streamsets.pipeline.lib.parser.DataParserException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+
+class Cells {
+  static Field parseCell(Cell cell) {
+    CellType cellType = cell.getCellTypeEnum();
+    if (cell.getCellTypeEnum().equals(CellType.FORMULA)) {
+      cellType = cell.getCachedFormulaResultTypeEnum();
+    }
+    switch (cellType) {
+      case STRING:
+        return Field.create(cell.getStringCellValue());
+      case NUMERIC:
+        return Field.create(cell.getNumericCellValue());
+      case BOOLEAN:
+        return Field.create(cell.getBooleanCellValue());
+      default:
+        try {
```

I think you can add this exception to the method signature and let it propagate upwards. The caller of the parser class is responsible for handling the exception.